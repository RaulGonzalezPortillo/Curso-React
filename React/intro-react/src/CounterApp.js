import React, { useState } from 'react'
import PropTypes from 'prop-types';
import './style.css';

export const CounterApp = ({value = 0}) => {
    const [counter, setCounter] = useState(value);

    const handleAdd = (e) => {
        setCounter (counter + 1);
        //setCounter ((c) => c + 1);
    };

    const handleSubstract = (e) => {
        if(counter > 0) {
            setCounter (counter - 1);
        }
    }

    const handleReset = (e) => {
        setCounter (value);
    }

    return (
        <>
            <h1>Counter</h1>
            <hr/>
            <h2>{counter}</h2>
            <div className="row">
                <button id="addButton" className="btn" onClick={handleAdd}>+</button>
                <button id="substractButton" className="btn" onClick={handleSubstract}>-</button>
                <button id="resetButton" className="btn" onClick={handleReset}>Reset</button>
            </div>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}