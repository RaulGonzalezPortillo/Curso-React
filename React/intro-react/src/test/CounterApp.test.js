import { shallow } from 'enzyme';
import { CounterApp } from '../CounterApp';

describe('Testing CounterApp Component', () => {
    test('Must show <CounterApp/> correctly', () => {
        const wrapper = shallow(<CounterApp/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Must show the counter sent by props', () => {
        const value = 100;
        const wrapper = shallow(<CounterApp value = {value} />);
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe(value.toString());
    });

    test('Must increment with + button', () => {
        const value = 99;
        const wrapper = shallow(<CounterApp value = {value} />);
        wrapper.find('#addButton').simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('100');
    })

    test('Must increment with - button', () => {
        const value = 88;
        const wrapper = shallow(<CounterApp value = {value} />);
        wrapper.find('#substractButton').simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('87');
    })

    test('Must reset with reset button', () => {
        const value = 69;
        const wrapper = shallow(<CounterApp value = {value} />);
        wrapper.find('#addButton').simulate('click');
        wrapper.find('#addButton').simulate('click');
        wrapper.find('#substractButton').simulate('click');
        wrapper.find('#addButton').simulate('click');
        wrapper.find('#addButton').simulate('click');
        wrapper.find('#substractButton').simulate('click');
        wrapper.find('#addButton').simulate('click');
        wrapper.find('#substractButton').simulate('click');
        wrapper.find('#resetButton').simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe(value.toString());
    })
});