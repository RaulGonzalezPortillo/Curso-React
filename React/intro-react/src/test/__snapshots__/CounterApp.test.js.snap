// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CounterApp Component Must show <CounterApp/> correctly 1`] = `
<Fragment>
  <h1>
    Counter
  </h1>
  <hr />
  <h2>
    0
  </h2>
  <div
    className="row"
  >
    <button
      className="btn"
      id="addButton"
      onClick={[Function]}
    >
      +
    </button>
    <button
      className="btn"
      id="substractButton"
      onClick={[Function]}
    >
      -
    </button>
    <button
      className="btn"
      id="resetButton"
      onClick={[Function]}
    >
      Reset
    </button>
  </div>
</Fragment>
`;
